src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -116,79 +118,92 @@ func logHostsStatus(log logrus.FieldLogger, hosts map[string]inventory_client.Ho
-		log.Infof("Checking if cluster nodes are ready. %d nodes remaining", len(hostsInProgressMap))
-		nodes, err := c.kc.ListNodes()
-		if err != nil {
-			log.WithError(err).Error("Failed to get list of nodes from k8s client")
 			continue
 		}
-		for _, node := range nodes.Items {
-			host, ok := hostsInProgressMap[strings.ToLower(node.Name)]
-			if !ok {
-				if _, ok := assistedNodesMap[strings.ToLower(node.Name)]; !ok {
-					log.Warnf("Node %s is not in inventory hosts", strings.ToLower(node.Name))
-				}
-
+		if common.IsK8sNodeIsReady(node) {
+			log.Infof("Found new ready node %s with inventory id %s, kubernetes id %s, updating its status to %s",
+				node.Name, host.Host.ID.String(), node.Status.NodeInfo.SystemUUID, models.HostStageDone)
+			if err := c.ic.UpdateHostInstallProgress(ctxReq, host.Host.ID.String(), models.HostStageDone, ""); err != nil {
+				log.WithError(err).Errorf("Failed to update node %s installation status", node.Name)
 				continue
 			}
-			if common.IsK8sNodeIsReady(node) {
-				log.Infof("Found new ready node %s with inventory id %s, kubernetes id %s, updating its status to %s",
-					node.Name, host.Host.ID.String(), node.Status.NodeInfo.SystemUUID, models.HostStageDone)
-				if err := c.ic.UpdateHostInstallProgress(ctx, host.Host.ID.String(), models.HostStageDone, ""); err != nil {
-					log.WithError(err).Errorf("Failed to update node %s installation status", node.Name)
-					continue
-				}
-			} else if host.Host.Progress.CurrentStage == models.HostStageConfiguring {
-				log.Infof("Found new joined node %s with inventory id %s, kubernetes id %s, updating its status to %s",
-					node.Name, host.Host.ID.String(), node.Status.NodeInfo.SystemUUID, models.HostStageJoined)
-				if err := c.ic.UpdateHostInstallProgress(ctx, host.Host.ID.String(), models.HostStageJoined, ""); err != nil {
-					log.WithError(err).Errorf("Failed to update node %s installation status", node.Name)
-					continue
-				}
+		} else if host.Host.Progress.CurrentStage == models.HostStageConfiguring {
+			log.Infof("Found new joined node %s with inventory id %s, kubernetes id %s, updating its status to %s",
+				node.Name, host.Host.ID.String(), node.Status.NodeInfo.SystemUUID, models.HostStageJoined)
+			if err := c.ic.UpdateHostInstallProgress(ctxReq, host.Host.ID.String(), models.HostStageJoined, ""); err != nil {
+				log.WithError(err).Errorf("Failed to update node %s installation status", node.Name)
+				continue
 			}
 		}
-		c.updateConfiguringStatusIfNeeded(assistedNodesMap)
 	}
-	c.log.Infof("Done waiting for all the nodes. Nodes in error status: %d\n", hostsInError)
+	c.updateConfiguringStatusIfNeeded(assistedNodesMap)
+	return false
 }
 
 func (c *controller) HackDNSAddressConflict(wg *sync.WaitGroup) {
 
 	c.log.Infof("Making sure service %s can reserve the .10 address", dnsServiceName)
 
-	defer wg.Done()
+	defer func() {
+		c.log.Infof("HackDNSAddressConflict finished")
```

Comment:
Most chances is that these shutdown logs will not show in the log file because the log routine is terminated as well

Reply:
it is made for checking logs inside pod. It is helpful for local debug or when you have connection to cluster

Reply:
most of them will be part of the logs. I think only upload logs will be missed

Reply:
It will depend on the random order of shutdown

Reply:
in any case we will wait for upload logs that will takes some time to finish, so we will have this log line in 90% of the cases. Other 10% will give better viability when we have connection to cluster